string_validator.rb

```diff
@@ -0,0 +1,35 @@
+class StringValidator
+
+  PAIRS = {
+    "]" => "[",
+    ")" => "(",
+    "}" => "{"
+  }
+
+  def validate(string)
```

Comment:
This is the first recursive approach I've seen. It's well done, but how would you do this iteratively? Would it make for a cleaner solution?